tests/e2e/gc_block_resize_retain_policy.go

```diff
@@ -725,6 +741,325 @@ var _ = ginkgo.Describe("[csi-guest] Volume Expansion Tests with reclaimation po
+	//     4. verify GC1 PVC is removed but SVC PV, PVC and GC1 PV still exist.
+	//     5. delete GC1 PV.  SVC PV, PVC still exist.
+	//     6. Create a new GC GC2.
+	//     7. create SC in GC1 similar to the SC created in step 1 but with reclaim
+	//        policy set to delete.
+	//     8. Create new PV in GC2 using the SVC PVC from step 5 and SC created in
+	//        step 7.
+	//     9. create new  PVC in GC2 using PV created in step 8.
+	//    10. verify a new PVC API object is created.
+	//    11. Create POD
+	//    12. Trigger online volume expansion on GC2 PVC
+	//    13, Check CNS querry for newly updated size
+	//    14. Check using CNS query that size has got updated to what was used in step 12
+	//    15. Verify size of PVs in SVC and GC to same as the one used in the step 12
+	//        delete PVC created in step 9.
+	//    16. delete SC created in step 1 and step 7.
+	//    17. delete GC2.
+
+	ginkgo.It("online volume expansion-PV with reclaim policy retain can be resized when used in a fresh GC", func() {
+		var err error
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		curtime := time.Now().Unix()
+		randomValue := rand.Int()
+		val := strconv.FormatInt(int64(randomValue), 10)
+		val = string(val[1:3])
+		curtimestring := strconv.FormatInt(curtime, 10)
+		svpvcName := "cns-pvc-" + curtimestring + val
+		framework.Logf("pvc name :%s", svpvcName)
+		namespace = getNamespaceToRunTests(f)
+
+		newGcKubconfigPath := os.Getenv("NEW_GUEST_CLUSTER_KUBE_CONFIG")
+		if newGcKubconfigPath == "" {
+			ginkgo.Skip("Env NEW_GUEST_CLUSTER_KUBE_CONFIG is missing")
+		}
+		clientNewGc, err = k8s.CreateKubernetesClientFromConfig(newGcKubconfigPath)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+			fmt.Sprintf("Error creating k8s client with %v: %v", newGcKubconfigPath, err))
+		ginkgo.By("Creating namespace on second GC")
+		ns, err := framework.CreateTestingNS(f.BaseName, clientNewGc, map[string]string{
+			"e2e-framework": f.BaseName,
+		})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Error creating namespace on second GC")
+		f.AddNamespacesToDelete(ns)
+
+		namespaceNewGC = ns.Name
+		framework.Logf("Created namespace on second GC %v", namespaceNewGC)
+		defer func() {
+			err := clientNewGc.CoreV1().Namespaces().Delete(ctx, namespaceNewGC, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Getting ready nodes on GC 2")
+		nodeList, err := fnodes.GetReadySchedulableNodes(clientNewGc)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(len(nodeList.Items)).NotTo(gomega.BeZero(), "Unable to find ready and schedulable Node in new GC")
+
+		_, storageclass, profileID := staticProvisioningPreSetUpUtil(ctx, f, client, storagePolicyName)
+
+		// Get supvervisor cluster client.
+		svcClient, svNamespace := getSvcClientAndNamespace()
+
+		ginkgo.By("Creating FCD (CNS Volume)")
+		fcdID, err := e2eVSphere.createFCDwithValidProfileID(ctx,
```

Comment:
we should cleanup this fcd as well, since we are using `Retain` policy

Reply:
done